server/resp/src/test/java/org/infinispan/server/resp/RespSetCommandsTest.java

```diff
@@ -140,6 +142,33 @@ public void testSintercard() {
       // SINTERCARD on an existing key that contains a List, not a Set!
       // Create a List
       assertWrongType(() -> redis.rpush("listleads", "tristan"), () -> redis.sintercard("listleads", "william"));
+
+      CustomStringCommands command = CustomStringCommands.instance(redisConnection);
+
+      assertThatThrownBy(() -> {
+         command.sintercard5Args("0".getBytes(),"a".getBytes(),"b".getBytes(),"c".getBytes(),"d".getBytes());
+      }).isInstanceOf(RedisCommandExecutionException.class)
+            .hasMessageStartingWith("ERR numkeys");
+
+      assertThatThrownBy(() -> {
+         command.sintercard5Args("notnum".getBytes(),"a".getBytes(),"b".getBytes(),"c".getBytes(),"d".getBytes());
+      }).isInstanceOf(RedisCommandExecutionException.class)
+            .hasMessageStartingWith("ERR numkeys");
+
+      assertThatThrownBy(() -> {
+         command.sintercard5Args("5".getBytes(),"a".getBytes(),"b".getBytes(),"c".getBytes(),"d".getBytes());
+      }).isInstanceOf(RedisCommandExecutionException.class)
+            .hasMessageStartingWith("ERR Number of keys");
+
+      assertThatThrownBy(() -> {
+         command.sintercard5Args("1".getBytes(),"a".getBytes(),"b".getBytes(),"c".getBytes(),"d".getBytes());
+      }).isInstanceOf(RedisCommandExecutionException.class)
+            .hasMessageStartingWith("ERR syntax error");
+
+      assertThatThrownBy(() -> {
+         command.sintercard5Args("1".getBytes(),"a".getBytes(),"b".getBytes(),"LIMIT".getBytes(),"-1".getBytes());
+      }).isInstanceOf(RedisCommandExecutionException.class)
+            .hasMessageStartingWith("ERR syntax error");
```

Comment:
Shouldn't this one receive the limit error message?

Reply:
the number of keys is checked before and results in an syntax error msg.
I changed the test a little bit and also added a test to check limit error message.